web/modules/custom/dbcdk_community/src/Form/ProfileEditForm.php

```diff
@@ -139,6 +161,18 @@ public function buildForm(array $form, FormStateInterface $form_state, $username
       '#date_date_format' => self::DATE_FORMAT,
     ];
 
+    $form['roles'] = [
+      '#type' => 'checkboxes',
+      '#title' => $this->t('Roles'),
+      '#default_value' => array_map(function($role) {
+        return (int) $role->getId();
+      }, $this->profile->communityRoles),
+      '#options' => array_reduce($this->communityRoles, function($result, $role) {
+        $result[(int) $role->getId()] = (string) $role->getName();
+        return $result;
```

Comment:
Using `array_reduce` to transform an keyed array to another was a bit surprising to me. I always saw the return value of this as a primitive (product of all values, all strings joined etc.) but I guess this works as well :+1: .